packages/core/src/locales/fi.ts

```diff
@@ -0,0 +1,197 @@
+import type { $ZodStringFormats } from "../checks.js";
+import type * as errors from "../errors.js";
+import type { $ZodTypeDef } from "../schemas.js";
+import * as util from "../util.js";
+
+const Sizable: Record<string, { unit: string; subject: string }> = {
+  string: { unit: "merkkiä", subject: "merkkijonon" },
+  file: { unit: "tavua", subject: "tiedoston" },
+  array: { unit: "alkiota", subject: "listan" },
+  set: { unit: "alkiota", subject: "joukon" },
+  number: { unit: "", subject: "luvun" },
+  bigint: { unit: "", subject: "suuren kokonaisluvun" },
+  int: { unit: "", subject: "kokonaisluvun" },
+  date: { unit: "", subject: "päivämäärän" },
+};
+
+function getSizing(origin: string): { unit: string; subject: string } | null {
+  return Sizable[origin] ?? null;
+}
+
+const TypeNames: { [k in $ZodTypeDef["type"] | (string & {})]?: string } = {
+  string: "merkkijono",
+  number: "luku",
+  boolean: "totuusarvo",
+  bigint: "suuri kokonaisluku",
+  symbol: "symboli",
+  null: "tyhjä",
+  undefined: "määrittelemätön",
+  date: "päivämäärä",
+  object: "objekti",
+  file: "tiedosto",
+  array: "lista",
+  map: "hakemisto",
+  set: "joukko",
+  nan: "epäluku",
+  promise: "lupaus",
+};
+
+function getTypeName(type: string): string {
+  return TypeNames[type] ?? type;
+}
+
+export const parsedType = (data: any): string => {
+  const t = typeof data;
+
+  switch (t) {
+    case "number": {
+      return Number.isNaN(data) ? "epäluku" : "luku";
+    }
+    case "bigint": {
+      return Number.isNaN(data) ? "epäluku" : "suuri kokonaisluku";
+    }
+    case "boolean": {
+      return "totuusarvo";
+    }
+    case "symbol": {
+      return "symboli";
+    }
+    case "function": {
+      return "funktio";
+    }
+    case "string": {
+      return "merkkijono";
+    }
+    case "undefined": {
+      return "määrittelemätön";
+    }
+    case "object": {
+      if (Array.isArray(data)) {
+        return "lista";
+      }
+      if (data === null) {
+        return "tyhjä";
+      }
+      if (data instanceof Date) {
+        return "päivämäärä";
+      }
+      if (data instanceof Map) {
+        return "hakemisto";
+      }
+      if (data instanceof Set) {
+        return "joukko";
+      }
+      if (data instanceof File) {
+        return "tiedosto";
+      }
+      if (data instanceof Promise) {
+        return "lupaus";
```

Comment:
Haha, this is pretty tricky.
I might prefer this to be `"Promise"` actually... Idk.